lib/husky_musher/app.py

```diff
@@ -151,121 +65,75 @@ def main():
-
-        # This won't always be @uw.edu.
-        "email": environ.get("mail", ""),
-
-        # Given name will include any middle initial/name.  Both name fields
-        # will contain the preferred name parts, if set, otherwise the
-        # administrative name parts.
-        "core_participant_first_name": environ.get("givenName", ""),
-        "core_participant_last_name":  environ.get("surname", ""),
-
-        # Department is generally a colon-separated set of
-        # increasingly-specific labels, starting with the School.
-        "uw_school": environ.get("department", ""),
-
-        **extract_affiliation(environ),
-    }
-
-
-def extract_affiliation(environ: dict) -> Dict[str, str]:
-    """
-    Transforms a multi-value affiliation string into our REDCap fields.
-
-    Keys of the returned dict match those used by our REDCap project.
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;faculty;employee;alum"})
-    {'affiliation': 'faculty', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;student;staff"})
-    {'affiliation': 'student', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;faculty;student"})
-    {'affiliation': 'student', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;staff;alum"})
-    {'affiliation': 'staff', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;employee"})
-    {'affiliation': 'staff', 'affiliation_other': ''}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;affiliate;alum"})
-    {'affiliation': 'other', 'affiliation_other': 'affiliate;alum'}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member"})
-    {'affiliation': '', 'affiliation_other': ''}
-
-    >>> extract_affiliation({})
-    {'affiliation': '', 'affiliation_other': ''}
-    """
-    raw_affilations = environ.get("unscoped-affiliation", "")
-
-    # "Member" is uninteresting and uninformative; a generic catch-all.
-    # The empty string might arise from our fallback above.
-    affiliations = set(raw_affilations.split(";")) - {"member",""}
-
-    rules = [
-        ("student"  in affiliations,    {"affiliation": "student",  "affiliation_other": ""}),
-        ("faculty"  in affiliations,    {"affiliation": "faculty",  "affiliation_other": ""}),
-        ("staff"    in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        ("employee" in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        (len(affiliations) > 0,         {"affiliation": "other",    "affiliation_other": ";".join(sorted(affiliations))}),
-        (True,                          {"affiliation": "",         "affiliation_other": ""})]
-
-    return next(result for condition, result in rules if condition)
+    netid = request.form['netid'].lower().strip()
+
+    if not re.match(r'^[a-z][a-z0-9]{,7}$', netid):
```

Comment:
I wonder if we will run into cases with non-conforming netids - according to https://wiki.cac.washington.edu/display/infra/UW+NetID+Namespace we may get additional characters in valid administrative/shared/temporary netids

Reply:
I don't think we would want someone enrolling in HCT with their administrative or shared NetIDs. I did not consider temporary IDs. It's always possible for someone to re-enroll if they signed up with the wrong NetID (e.g. an admin account). 

If the temporary NetID contains characters invalid for a personal NetID, the kiosk staff could always manually proceed with the walk-up kiosk flow (as they're currently doing). Otherwise, we could have a button on the `invalid_netid.html` template that makes the kiosk staff double-check then proceeds anyway. Thoughts?

Reply:
Good points, they do still have the manual lookup option. How about we take the shortest route: add text to the invalid_netid.html to tell them to manually lookup if they are sure its correct, and please slack dev team to let us know (so we could potentially update our logic).

Reply:
Sure, I added this in a new commit.